functions.lua

```diff
@@ -503,9 +504,26 @@ function tp.tpc_send(sender, coordinates)
 		if minetest.check_player_privs(pname, {tp_tpc = true}) then
 			local protected = minetest.is_protected(target_coords, sender)
 			if protected then
-				if minetest.get_modpath("areas") then
+				-- Now, we can safely check if the `check_area_requests` option is enabled (see the `config.lua` file, lines 28-35)
+				if minetest.get_modpath("areas") and check_area_requests then
 					for _, area in pairs(areas:getAreasAtPos(target_coords)) do
 						if minetest.get_player_by_name(area.owner) then -- Check if area owners are online
+							-- Spam prevention
+							if spam_prevention[area.owner] == sender and not minetest.check_player_privs(sender, {tp_admin = true}) then
```

Comment:
If there's a similar code snippet somewhere, please offload the check and `minetest.after` call to a helper function.